### ISSUE TITLE
Consider writing a function to replace shared code across pages

### ISSUE THREAD

**Issue by Bergam0t** [BarnabyRumbold/HSMA#4]
Might be nice to see if you can turn this into a function that returns `co_eff_df` at the end. It looks like it's very similar code across the different pages, so it's more maintainable and less likely to introduce errors if you have a shared function for the purpose. 

(Could return several things if you wanted to be able to use some of the other things you calculate along the way like the accuracy and the model object itself - see https://python.hsma.co.uk/functions.html#returning-multiple-values)

You could replace the 'ATTENDED' in this example with a variable, and the same with the 'GENDER_DESC_' and the replacement value, so this could be used flexibly across multiple pages. 

You might want to set it up in a separate .py file and then import it into the different .py files throughout. 

```
# Split data
X = df.drop('ATTENDED',axis=1)
y = df['ATTENDED']

# Prepare data set for logistic regression - decide weights and clean data set (potentially in query)
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.25)

# Fit model 
model = LogisticRegression()
model.fit(X_train, y_train)

# Predict
y_pred_train = model.predict(X_train)
y_pred_test = model.predict(X_test)

# Predictions and accuracy test
accuracy_train = np.mean(y_pred_train == y_train)
accuracy_test = np.mean(y_pred_test == y_test)
co_eff = model.coef_[0]
co_eff_df = pd.DataFrame() # create empty DataFrame
co_eff_df['feature'] = list(X) # Get feature names from X
co_eff_df['co_eff'] = co_eff
co_eff_df['abs_co_eff'] = np.abs(co_eff)
co_eff_df.sort_values(by='abs_co_eff', ascending=False, inplace=True)
co_eff_df['feature'] = co_eff_df['feature'].str.replace("GENDER_DESC_","")
co_eff_df = co_eff_df.round(3)

```
